File: ruby/ch02/ex01/input.rb
class Hello

  attr_reader :name

  def initialize(name)
    @name = name
  end

  def say_hello
    puts create_output
  end

  private
  
  def create_output
    "Hello, #{name}, nice to meet you!"
  end
end

print "What is your name? "
Hello.new(gets.chomp).say_hello

File: elixir/ch02/ex01-saying-hello/input.exs
defmodule Greeter do
  def ask_and_greet do
    ask_name
    |> sanitize_input
    |> create_greet_message
    |> greet
  end

  def ask_name, do: IO.gets("What is your name? ")

  def sanitize_input(input), do: String.strip(input)

  def create_greet_message(name) do
    cond do
      name =~ ~r/^B.*$/ 
        -> "Hello, #{name}, nice to meet you!"
      _name 
        -> "Hello, beautiful stranger!" 
    end
  end

  def greet(name), do: IO.puts(name)
end

Greeter.ask_and_greet
